conjure-java-jersey-server/src/main/java/com/palantir/conjure/java/server/jersey/WebApplicationExceptionMapper.java

```diff
@@ -52,10 +52,10 @@
     public Response toResponseInner(WebApplicationException exception) {
         String errorInstanceId = UUID.randomUUID().toString();
 
-        if (exception.getResponse().getStatus() / 100 == 4 /* client error */) {
-            log.info("Error handling request", SafeArg.of("errorInstanceId", errorInstanceId), exception);
-        } else {
+        if (exception.getResponse().getStatusInfo().getFamily() == Response.Status.Family.SERVER_ERROR) {
             log.error("Error handling request", SafeArg.of("errorInstanceId", errorInstanceId), exception);
+        } else {
+            log.info("Error handling request", SafeArg.of("errorInstanceId", errorInstanceId), exception);
```

Comment:
I didn't change the behavior on the info logging here, but a case could be made we probably don't need the exception for non-5xx responses